apps/meteor/client/views/omnichannel/analytics/DateRangePicker.tsx

```diff
@@ -1,24 +1,36 @@
 import { Box, InputBox, Menu, Field } from '@rocket.chat/fuselage';
 import { useMutableCallback } from '@rocket.chat/fuselage-hooks';
 import { useTranslation } from '@rocket.chat/ui-contexts';
-import moment from 'moment';
+import type { ComponentProps, ReactElement } from 'react';
 import React, { useState, useMemo, useEffect } from 'react';
 
-const formatToDateInput = (date) => date.format('YYYY-MM-DD');
+const date = new Date();
 
-const todayDate = formatToDateInput(moment());
+const formatToDateInput = (date: Date): string => date.toISOString().slice(0, 10);
 
-const getMonthRange = (monthsToSubtractFromToday) => ({
-	start: formatToDateInput(moment().subtract(monthsToSubtractFromToday, 'month').date(1)),
-	end: formatToDateInput(monthsToSubtractFromToday === 0 ? moment() : moment().subtract(monthsToSubtractFromToday).date(0)),
-});
+const todayDate = formatToDateInput(date);
 
-const getWeekRange = (daysToSubtractFromStart, daysToSubtractFromEnd) => ({
-	start: formatToDateInput(moment().subtract(daysToSubtractFromStart, 'day')),
-	end: formatToDateInput(moment().subtract(daysToSubtractFromEnd, 'day')),
-});
+const getMonthRange = (monthsToSubtractFromToday: number): { start: string; end: string } => {
+	const date = new Date();
+	return {
+		start: formatToDateInput(new Date(date.getFullYear(), date.getMonth() - monthsToSubtractFromToday, 1)),
+		end: formatToDateInput(new Date(date.getFullYear(), date.getMonth() - monthsToSubtractFromToday + 1, 0)),
+	};
+};
+
+const getWeekRange = (daysToSubtractFromStart: number, daysToSubtractFromEnd: number): { start: string; end: string } => {
+	const date = new Date();
+	return {
+		start: formatToDateInput(new Date(date.getFullYear(), date.getMonth(), date.getDate() - daysToSubtractFromStart)),
+		end: formatToDateInput(new Date(date.getFullYear(), date.getMonth(), date.getDate() - daysToSubtractFromEnd)),
+	};
```

Comment:
What was the problem with moment? 👀 any FE guideline to not to use it?

Reply:
I was having problems with the type of date when I was using moment, so I search for examples about how to deal with dates on the repo and they are doing it this way. Do you think that I should turn back and use moment?